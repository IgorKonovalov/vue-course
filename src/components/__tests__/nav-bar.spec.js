import Vuex from 'vuex';
import { shallowMount, createLocalVue } from '@vue/test-utils';
import NavBar from '../nav-bar.vue';

const localVue = createLocalVue();
localVue.use(Vuex);

describe('NavBar', () => {
  let wrapper;
  let store;
  let loadUsersMock;

  beforeEach(() => {
    loadUsersMock = jest.fn();

    store = new Vuex.Store({
      modules: {
        account: {
          namespaced: true,
          state: {
            accountName: 'accountName',
            accountAvatar: 'accountAvatar'
          }
        },
        users: {
          namespaced: true,
          state: {},
          mutations: {
            LOAD_USERS: loadUsersMock
          }
        }
      }
    })

    wrapper = shallowMount(NavBar, { store, localVue });
  });

  afterEach(() => {
    wrapper.destroy();
  });

  it('renders correctly', () => {
    expect(wrapper.vm.$el).toMatchSnapshot()
  });
});
