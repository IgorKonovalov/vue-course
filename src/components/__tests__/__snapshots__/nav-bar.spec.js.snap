// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar renders correctly 1`] = `
<nav>
  <div
    class="nav-header"
  >
    <div
      class="text-input"
    >
      <input
        class="aqa-input-add-user"
        type="text"
      />
       
      <button
        class="btn-add aqa-button-add"
      >
        
        Add
      
      </button>
    </div>
     
    <div
      class="account"
    >
      <span
        class="name"
      >
        accountName
      </span>
       
      <img
        class="avatar"
        height="40"
        src="accountAvatar"
        width="40"
      />
    </div>
  </div>
</nav>
`;
